xml/hwbp_registers.xml

```diff
@@ -746,6 +749,12 @@
             \Rxepc or \RcsrDpc (depending on \FcsrMcontrolSixAction) must be set
             to the virtual address of the instruction that matched.
 
+            If an instruction partially executed, the reported timing is still
+            before. This might mean that when the instruction is executed again
+            it will perform some of the same stores again (e.g. \tt{cbo.zero}),
+            or the instruction might have a mechanism that avoids duplicating the
+            stores (e.g. vector stores and the vstart mechanism).
```

Comment:
This focuses on stores, but what about register results of an instruction that performs multiple memory accesses?

Must any and all register results be performed only if the instruction completes, i.e. before timing disallows any register modifications (that might, for example, prevent restartability of the instruction aupon return from the debug handler), and after timing of course requires all such modifications to have been performed.

And what about loads?  Are some allowed, like stores, to have been performed under before timing?  And what if some of those loads are to non-idempotent memory?

Lastly, cbo.zero is not the best of example since it architecturally performs a single block-size memory write.  (An implementation can choose to implement that as a series of smaller stores of some size, but the expected common implementation would be one memory write.)

It seems like this paragraph needs to cover read acesses, write accesses, and register writes.  Whether the statement is common to all three, or is a separate statement for register writes and for memory accesses, or maybe different for all three, all three need to be covered explicitly in some way so that peopel know what are  the bounds on allowed and disallowed behavior for before triggers and for after triggers.

Reply:
The first sentence is the important one. "If an instruction partially executed, the reported timing is still before." The rest is just example. I'll tweak the language to allow for more implementations. Do you have a suggestion for an instruction that performs multiple operations but that doesn't have a restart mechanism? I can't think of any.

Reply:
There's no obvious perfect statement one coudl make.  But one needs to inform users what is and isn't possible.  Obviously users prefer a more useful statement that will allow most instructions to be restartable.

For example (and I'm not trying to say that this is what should be specified - the TG needs to consider the pros and cons of whatever it would choose to specify), one could say that some of the memory accesses (loads and stores) may have been performed but no registers will have been modified.  And one should also note that while most instructions will still always be restartable, some may not be (and accesses to non-idempotent memory may be repeated).

Reply:
I've rewritten almost all of the mcontrol6 `before` timing. Can you see if this is more clear?